src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -264,6 +264,19 @@ public override void AddPackage(IPackage package)
                 throw new InvalidOperationException(message);
             }
 
+            if (RetentionRule > 0)
+            {
+                var packages = _expandedPackageRepository.GetPackages();
+                var samePackages = packages.Where(p => p.Id == package.Id).OrderBy(p => p.Version).ToList();
```

Comment:
There are UTs for this class here: [https://github.com/NuGet/NuGet.Server/blob/master/test/NuGet.Server.Tests/ServerPackageRepositoryTest.cs](https://github.com/NuGet/NuGet.Server/blob/master/test/NuGet.Server.Tests/ServerPackageRepositoryTest.cs)
Can you please add tests for this feature?